packages/block-editor/src/components/iframe/index.js

```diff
@@ -161,15 +175,19 @@ function Iframe( { contentRef, children, head, headHTML, ...props }, ref ) {
 				contentRef.current = body;
 			}
 
-			setHead( contentDocument, headHTML );
 			setBodyClassName( contentDocument );
-			styleSheetsCompat( contentDocument );
 			bubbleEvents( contentDocument );
 			setBodyClassName( contentDocument );
 			setIframeDocument( contentDocument );
 			clearerRef( documentElement );
 			clearerRef( body );
 
+			scripts.reduce(
+				( promise, script ) =>
+					promise.then( () => loadScript( contentDocument, script ) ),
```

Comment:
Yes, otherwise a script might load before a dependency has loaded? It would be nice to load in parallel, but evaluate in series. Not sure how that would work though.

I doesn't block rendering so it seems fine to me. We can polish this later on .

Reply:
> Yes, otherwise a script might load before a dependency has loaded?

Yeah, you're right. Parallelising isn't impossible, but we'd need a careful method to determine dependencies.